app/src/main/java/com/github/geohunt/app/model/database/Database.kt

```diff
@@ -47,6 +43,24 @@ interface Database {
      */
     fun getNearbyChallenge(location: Location): Task<List<Challenge>>
 
+    /**
+     * Returns the followers of the user with the given user id.
+     *
+     * @param uid The user ID.
+     * @return A map where keys that are mapped to true indicates that it is a follower.
+     */
+    fun getFollowersOf(uid: String): Task<Map<String, Boolean>>
```

Comment:
Your function currently returns a map that includes both user IDs and boolean values. The boolean value indicates whether a given user is a follower of the input user.

However, I recommend modifying the function to return a single list containing only the user IDs of the input user's followers. This will make the output smaller and easier to iterate through. This approach would be better both from the perspective of code clarity and performance.

Reply:
As indicated in the PR description, the usage of `Map`s is a technicality due to the nature of the database being NoSQL. `List`s are not well represented in the database, because they are essentially encoded as objects whose keys are integers. They may thus lead to subtle bugs, e.g. when you remove elements (see for more explanation the [link posted above](https://firebase.blog/posts/2014/04/best-practices-arrays-in-firebase)). As it is a method of the `FirebaseDatabase`, `getFollowersOf` only reflects what is in database, i.e. a map.

I do not agree with your last 2 sentences: there is no performance penalty for using `Map`, nor is the output bigger, because only relevant keys are stored (by default the map returns `false` for any key it doesn't recognize). Furthermore, it is easy from the `Map` to get the `List` of followers, because that is exactly the set of keys whose associated value is `true`.

To make things clearer for the UI, I can change the API to return a `List<FirebaseUserRef>`, *but* I think that a __very big warning__ should be added to the method's documentation, or otherwise there will be bugs because someone will push a `List` to the database, instead of a `Map`.